include/net/multicell_location.h

```diff
@@ -41,28 +41,39 @@ enum multicell_service {
 	MULTICELL_SERVICE_POLTE
 };
 
+/** Cellular positioning input parameters. */
+struct multicell_location_params {
+	/** Cellular positioning service to be used. */
+	enum multicell_service service;
+	/** Neighbor cell data. */
+	const struct lte_lc_cells_info *cell_data;
+	/**
+	 * @brief Timeout (in seconds) of how long the cellular positioning procedure can take.
+	 * Zero means that the timer is disabled.
+	 */
+	uint16_t timeout;
+};
+
 /**
  * @brief Send a request for location based on cell measurements to the
  *        selected location service.
  *
  * @note This function will block the calling thread until a response
- *       is received from the location service.
+ *       is received from the location service, or timeout has elapsed.
  *
  * @note Certificate must be provisioned before a request can be sent,
  *       @ref multicell_location_provision_certificate.
  *
- * @param[in] service Cellular positioning service to be used.
- * @param[in] cell_data Pointer to neighbor cell data.
- * @param[out] location Pointer to location.
+ * @param[in] params Cellular positioning parameters to be used.
+ * @param[out] location Location.
  *
  * @return 0 on success, or negative error code on failure.
  * @retval -ENOENT No cellular cells found from cell_data. I.e., even current cell
  *                 is not available and there is no cellular connectivity.
  * @retval -ENOMEM Out of memory.
  * @retval -ENOMSG Parsing response from the location service failed.
  */
-int multicell_location_get(enum multicell_service service,
-			   const struct lte_lc_cells_info *cell_data,
+int multicell_location_get(struct multicell_location_params *params,
```

Comment:
Could be `const`, the parameters should be input only.

Reply:
Changed